src/os/bluestore/BlueStore.cc

```diff
@@ -2768,7 +2823,29 @@ void BlueStore::Onode::flush()
 }
 
 // =======================================================
-
+// WriteContext
+ 
+/// Checks for writes to the same pextent within a blob
+bool BlueStore::WriteContext::has_conflict(
+  BlobRef b,
+  uint64_t loffs,
+  uint64_t blob_len)
+{
+  loffs = P2ALIGN(loffs, blob_len);
```

Comment:
This is allocated blob length and hence it has min_alloc_size granularity. That's supposed to be power of 2 IMO.
Moreover currently it's used from do_write_small only that always allocates single AU. Will add an assert though...